azurecosmos/src/main/java/com/yahoo/ycsb/db/AzureCosmosClient.java

```diff
@@ -0,0 +1,454 @@
+  private static final boolean DEFAULT_INCLUDE_EXCEPTION_STACK_IN_LOG = false;
+
+  private static final int NS_IN_US = 1000;
+  private static final String NA_STRING = "N/A";
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(AzureCosmosClient.class);
+
+  /**
+   * Count the number of times initialized to teardown on the last
+   * {@link #cleanup()}.
+   */
+  private static final AtomicInteger INIT_COUNT = new AtomicInteger(0);
+
+  private static DocumentClient client;
+
+  private String queryText;
+  private String databaseName;
+  private boolean useSinglePartitionCollection;
+  private boolean useUpsert;
+  private boolean useHashQueryForScan;
+  private int maxDegreeOfParallelismForQuery;
+  private boolean includeExceptionStackInLog;
+
+  @Override
+  public synchronized void init() throws DBException {
+    INIT_COUNT.incrementAndGet();
+    if (client != null) {
+      return;
+    }
+
+    initAzureCosmosClient();
+  }
+
+  private void initAzureCosmosClient() throws DBException {
+    // Connection properties
+    String primaryKey = this.getStringProperty("azurecosmos.primaryKey", null);
+    if (StringUtils.isEmpty(primaryKey)) {
+      throw new DBException("Missing primary key required to connect to the database.");
+    }
+
+    String uri = getProperties().getProperty("azurecosmos.uri", null);
+    if (StringUtils.isEmpty(uri)) {
+      throw new DBException("Missing uri required to connect to the database.");
+    }
+
+    this.useSinglePartitionCollection = this.getBooleanProperty(
+            "azurecosmos.useSinglePartitionCollection",
+            DEFAULT_USE_SINGLE_PARTITION_COLLECTION);
+
+    this.useUpsert = this.getBooleanProperty(
+            "azurecosmos.useUpsert",
+            DEFAULT_USE_UPSERT);
+
+    this.useHashQueryForScan = this.getBooleanProperty(
+            "azurecosmos.useHashQueryForScan",
+            DEFAULT_USE_HASH_QUERY_FOR_SCAN);
+
+    this.databaseName = this.getStringProperty(
+            "azurecosmos.databaseName",
+            DEFAULT_DATABASE_NAME);
+
+    this.maxDegreeOfParallelismForQuery = this.getIntProperty(
+            "azurecosmos.maxDegreeOfParallelismForQuery",
+            DEFAULT_MAX_DEGREE_OF_PARALLELISM_FOR_QUERY);
+
+    this.includeExceptionStackInLog = this.getBooleanProperty(
+            "azurecosmos.includeExceptionStackInLog",
+            DEFAULT_INCLUDE_EXCEPTION_STACK_IN_LOG);
+
+    ConsistencyLevel consistencyLevel = ConsistencyLevel.valueOf(this.getStringProperty(
+            "azurecosmos.consistencyLevel",
+            DEFAULT_CONSISTENCY_LEVEL));
+    String connectionModeString = this.getStringProperty(
+            "azurecosmos.connectionMode",
+            DEFAULT_CONNECTION_MODE);
+
+    ConnectionPolicy connectionPolicy = new ConnectionPolicy();
+    connectionPolicy.setEnableEndpointDiscovery(false);
+    connectionPolicy.setConnectionMode(ConnectionMode.valueOf(connectionModeString));
```

Comment:
formatting of this block of code seems to be off a little bit, it's not aligned with the code above.

Reply:
fixed